crates/engine/tree/src/tree/mod.rs

```diff
@@ -518,20 +518,23 @@ where
         loop {
             match self.try_recv_engine_message() {
                 Ok(Some(msg)) => {
-                    self.on_engine_message(msg);
+                    if let Err(fatal) = self.on_engine_message(msg) {
+                        error!(target: "engine", %fatal, "insert block fatal error");
+                        break;
+                    }
                 }
                 Ok(None) => {
                     debug!(target: "engine", "received no engine message for some time, while waiting for persistence task to complete");
                 }
                 Err(_err) => {
                     error!(target: "engine", "Engine channel disconnected");
-                    return
+                    return;
                 }
             }
 
             if let Err(err) = self.advance_persistence() {
                 error!(target: "engine", %err, "Advancing persistence failed");
-                break
+                break;
```

Comment:
let's `return` here, they should be equivalent but this is just to stay consistent

Reply:
gotcha